worlds/messenger/client_setup.py

```diff
@@ -0,0 +1,168 @@
+        raise RuntimeError("Failed to install Courier")
+
+    def install_mod() -> None:
+        """Installs latest version of the mod"""
+        assets = request_data(MOD_URL)["assets"]
+        if len(assets) == 1:
+            release_url = assets[0]["browser_download_url"]
+        else:
+            for asset in assets:
+                if "TheMessengerRandomizerAP" in asset["name"]:
+                    release_url = asset["browser_download_url"]
+                    break
+            else:
+                messagebox("Failure", "Failed to find latest mod download", True)
+                raise RuntimeError("Failed to install Mod")
+
+        mod_folder = os.path.join(game_folder, "Mods")
+        os.makedirs(mod_folder, exist_ok=True)
+        with urllib.request.urlopen(release_url) as download:
+            with ZipFile(io.BytesIO(download.read()), "r") as zf:
+                for member in zf.infolist():
+                    zf.extract(member, path=mod_folder)
+
+        messagebox("Success!", "Latest mod successfully installed!")
+
+    def available_mod_update(latest_version: str) -> bool:
+        """Check if there's an available update"""
+        latest_version = latest_version.lstrip("v")
+        toml_path = os.path.join(game_folder, "Mods", "TheMessengerRandomizerAP", "courier.toml")
+        with open(toml_path, "r") as f:
+            installed_version = f.read().splitlines()[1].strip("version = \"")
+
+        logging.info(f"Installed version: {installed_version}. Latest version: {latest_version}")
+        # one of the alpha builds
+        return "alpha" in latest_version or tuplize_version(latest_version) > tuplize_version(installed_version)
+
+    from . import MessengerWorld
+    game_folder = os.path.dirname(MessengerWorld.settings.game_path)
+    working_directory = os.getcwd()
+    if not courier_installed():
+        should_install = askyesnocancel("Install Courier",
+                                        "No Courier installation detected. Would you like to install now?")
+        if not should_install:
+            return
+        logging.info("Installing Courier")
+        install_courier()
+    if not mod_installed():
+        should_install = askyesnocancel("Install Mod",
+                                        "No randomizer mod detected. Would you like to install now?")
+        if not should_install:
+            return
+        logging.info("Installing Mod")
+        install_mod()
+    else:
+        latest = request_data(MOD_URL)["tag_name"]
+        if available_mod_update(latest):
+            should_update = askyesnocancel("Update Mod",
+                                           f"New mod version detected. Would you like to update to {latest} now?")
+            if should_update:
+                logging.info("Updating mod")
+                install_mod()
+            elif should_update is None:
+                return
+    if is_linux:
```

Comment:
i had no idea if it would also work on mac which is why i was explicit about not supporting it, but if it should work the same way i'll add that in. for linux https://discord.com/channels/731205301247803413/1090822939697500243/1215326568348389426

Reply:
It's hard to say how long stuff is gonna work on mac, with mono being deprecated, and also I have no idea if courier actually works, but both [mono](https://formulae.brew.sh/formula/mono) as well as steam is available on mac.